src/Stream/FileWriter.cpp

```diff
@@ -38,6 +38,10 @@ namespace Stream
 	FileWriter::FileWriter(const std::string& filename, OpenMode openMode) :
 		filename(filename)
 	{
+		if (XFile::IsDirectory(filename) && XFile::PathExists(filename)) {
```

Comment:
My gut here is that `IsDirectory` should be sufficient. To get `true` from `IsDirectory`, I assume the path either needs to exist, or it was the empty string `""`.

We might want to throw with a custom error message for an empty `filename`. I'm not sure what error message that would give currently.

Reply:
Custom error message for filename is now incorporated. I still need to test IsDirectory as mentioned to see actual behaviour. 

FileWriter testing would probably benefit from using GoogleMock.